src/terraform/devcontainer-feature.json

```diff
@@ -47,6 +47,11 @@
             "type": "boolean",
             "default": false,
             "description": "Install terraform-docs, a utility to generate documentation from Terraform modules"
+        },
+        "http_proxy": {
+            "type": "string",
```

Comment:
Once you change the case, you'd need to add the following to https://github.com/devcontainers/features/blob/main/src/terraform/install.sh

> HTTP_PROXY=${HTTPPROXY:-""}

Then consume `$HTTP_PROXY` in the install.sh file

Reply:
This is the result of testing.

* Test using devcontainers features test in internet direct environment .. OK
* Test using devcontainers features test in proxy environment .. NG
* Manually start the container and run install.sh .. NG
* Manually start a container and make changes to install.sh and run it in the manually started container .. OK

I think I don't fully understand the features option.

Append the log about the failed test to a comment.

Reply:
Test using devcontainers features test in proxy environment.
I don't think this fix is ​​the cause of the test failure.

```
node ➜ /workspaces/features (terraform-gpg) $ devcontainer features test --features terraform --base-image ubuntu

┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
|    Dev Container Features   |   
│           v0.31.0           │
└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘

>  baseImage:         ubuntu
>  Target Folder:     /workspaces/features
>  features:          terraform
>  workspaceFolder:   /tmp/devcontainercli/container-features-test/1677461366608

⏳ Building test container...

[1 ms] @devcontainers/cli 0.31.0. Node.js v18.14.0. linux 5.15.0-1028-aws x64.
[273 ms] Start: Run: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-node/container-features/0.31.0-1677461367478 --build-arg _DEV_CONTAINERS_BASE_IMAGE=ubuntu --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-1677461366608-1d97999891254253382c9d6e08cb3cf9-features -f /tmp/devcontainercli-node/container-features/0.31.0-1677461367478/Dockerfile.extended /tmp/devcontainercli-node/empty-folder
[+] Building 41.4s (15/15) FINISHED                                             
 => [internal] load build definition from Dockerfile.extended              0.0s
 => => transferring dockerfile: 1.60kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4               1.6s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531b  0.0s
 => [internal] load .dockerignore                                          0.0s
 => [internal] load build definition from Dockerfile.extended              0.0s
 => [internal] load metadata for docker.io/library/ubuntu:latest           0.0s
 => [context dev_containers_feature_content_source] load .dockerignore     0.0s
 => => transferring dev_containers_feature_content_source: 2B              0.0s
 => [context dev_containers_feature_content_source] load from client       0.0s
 => => transferring dev_containers_feature_content_source: 22.79kB         0.0s
 => [dev_containers_feature_content_normalize 1/3] FROM docker.io/library  0.0s
 => CACHED [dev_containers_feature_content_normalize 2/3] COPY --from=dev  0.0s
 => CACHED [dev_containers_feature_content_normalize 3/3] RUN chmod -R 07  0.0s
 => CACHED [dev_containers_target_stage 2/4] COPY --from=dev_containers_f  0.0s
 => CACHED [dev_containers_target_stage 3/4] RUN echo "_CONTAINER_USER_HO  0.0s
 => ERROR [dev_containers_target_stage 4/4] RUN --mount=type=bind,from=d  39.3s
------                                                                          
 > [dev_containers_target_stage 4/4] RUN --mount=type=bind,from=dev_containers_feature_content_source,source=terraform_1,target=/tmp/build-features-src/terraform_1     cp -ar /tmp/build-features-src/terraform_1 /tmp/build-features/  && chmod -R 0700 /tmp/build-features/terraform_1  && cd /tmp/build-features/terraform_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/build-features/terraform_1:
#0 0.634 ===========================================================================
#0 0.634 Feature       : ./terraform
#0 0.634 Description   : Installs the Terraform CLI and optionally TFLint and Terragrunt. Auto-detects latest version and installs needed dependencies.
#0 0.634 Id            : Unknown
#0 0.634 Version       : 1.3.0
#0 0.634 Documentation : https://github.com/devcontainers/features/tree/main/src/terraform
#0 0.634 Options       :
#0 0.634     VERSION="latest"
#0 0.634     TFLINT="latest"
#0 0.634     TERRAGRUNT="latest"
#0 0.634     INSTALLSENTINEL="false"
#0 0.634     INSTALLTFSEC="false"
#0 0.634     INSTALLTERRAFORMDOCS="false"
#0 0.634     HTTPPROXY=""
#0 0.634 ===========================================================================
#0 0.675 find: '/var/lib/apt/lists/*': No such file or directory
#0 0.676 Running apt-get update...
#0 31.83 Ign:1 http://security.ubuntu.com/ubuntu jammy-security InRelease
#0 32.07 Ign:2 http://archive.ubuntu.com/ubuntu jammy InRelease
#0 32.08 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
#0 32.10 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
#0 32.85 Ign:1 http://security.ubuntu.com/ubuntu jammy-security InRelease
#0 33.08 Ign:2 http://archive.ubuntu.com/ubuntu jammy InRelease
#0 33.10 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
#0 33.11 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
#0 34.86 Ign:1 http://security.ubuntu.com/ubuntu jammy-security InRelease
#0 35.10 Ign:2 http://archive.ubuntu.com/ubuntu jammy InRelease
#0 35.11 Ign:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
#0 35.12 Ign:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
#0 38.87 Err:1 http://security.ubuntu.com/ubuntu jammy-security InRelease
#0 38.87   Could not connect to security.ubuntu.com:80 (91.189.91.38), connection timed out Could not connect to security.ubuntu.com:80 (91.189.91.39), connection timed out Could not connect to security.ubuntu.com:80 (185.125.190.36), connection timed out Could not connect to security.ubuntu.com:80 (185.125.190.39), connection timed out
#0 39.11 Err:2 http://archive.ubuntu.com/ubuntu jammy InRelease
#0 39.11   Could not connect to archive.ubuntu.com:80 (185.125.190.36), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.91.38), connection timed out Could not connect to archive.ubuntu.com:80 (185.125.190.39), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.91.39), connection timed out
#0 39.12 Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
#0 39.12   Unable to connect to archive.ubuntu.com:80:
#0 39.13 Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
#0 39.13   Unable to connect to archive.ubuntu.com:80:
#0 39.15 Reading package lists...
#0 39.19 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease  Could not connect to archive.ubuntu.com:80 (185.125.190.36), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.91.38), connection timed out Could not connect to archive.ubuntu.com:80 (185.125.190.39), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.91.39), connection timed out
#0 39.19 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease  Unable to connect to archive.ubuntu.com:80:
#0 39.19 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease  Unable to connect to archive.ubuntu.com:80:
#0 39.19 W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease  Could not connect to security.ubuntu.com:80 (91.189.91.38), connection timed out Could not connect to security.ubuntu.com:80 (91.189.91.39), connection timed out Could not connect to security.ubuntu.com:80 (185.125.190.36), connection timed out Could not connect to security.ubuntu.com:80 (185.125.190.39), connection timed out
#0 39.19 W: Some index files failed to download. They have been ignored, or old ones used instead.
#0 39.23 Reading package lists...
#0 39.27 Building dependency tree...
#0 39.27 Reading state information...
#0 39.27 Package ca-certificates is not available, but is referred to by another package.
#0 39.27 This may mean that the package is missing, has been obsoleted, or
#0 39.27 is only available from another source
#0 39.27 
#0 39.27 Package gnupg2 is not available, but is referred to by another package.
#0 39.27 This may mean that the package is missing, has been obsoleted, or
#0 39.27 is only available from another source
#0 39.27 However the following packages replace it:
#0 39.27   gpgv
#0 39.27 
#0 39.28 E: Unable to locate package curl
#0 39.28 E: Package 'ca-certificates' has no installation candidate
#0 39.28 E: Package 'gnupg2' has no installation candidate
#0 39.28 E: Unable to locate package dirmngr
#0 39.28 E: Unable to locate package unzip
#0 39.28 ERROR: Feature "./terraform" (Unknown) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/terraform for help troubleshooting this error.
------
ERROR: failed to solve: executor failed running [/bin/sh -c cp -ar /tmp/build-features-src/terraform_1 /tmp/build-features/  && chmod -R 0700 /tmp/build-features/terraform_1  && cd /tmp/build-features/terraform_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/build-features/terraform_1]: exit code: 100
[-] Failed to launch container:

Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-node/container-features/0.31.0-1677461367478 --build-arg _DEV_CONTAINERS_BASE_IMAGE=ubuntu --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-1677461366608-1d97999891254253382c9d6e08cb3cf9-features -f /tmp/devcontainercli-node/container-features/0.31.0-1677461367478/Dockerfile.extended /tmp/devcontainercli-node/empty-folder
node ➜ /workspaces/features (terraform-gpg) $
```

Reply:
Manually start the container and run install.sh

```
vscode ➜ /workspaces/vscode-sandbox (devcon) $ echo $HTTP_PROXY | sed -e 's/'$MASK'/xxx/'
http://proxy.xxx.co.jp:8080
vscode ➜ /workspaces/vscode-sandbox (devcon) $ grep image .devcontainer/devcontainer.json
        "image": "mcr.microsoft.com/devcontainers/base:jammy"
vscode ➜ /workspaces/vscode-sandbox (devcon) $ cd
vscode ➜ ~ $ git clone https://github.com/ksaito1125/features.git
Cloning into 'features'...
remote: Enumerating objects: 3970, done.
remote: Counting objects: 100% (314/314), done.
remote: Compressing objects: 100% (181/181), done.
remote: Total 3970 (delta 172), reused 240 (delta 131), pack-reused 3656
Receiving objects: 100% (3970/3970), 4.03 MiB | 11.45 MiB/s, done.
Resolving deltas: 100% (2331/2331), done.
vscode ➜ ~ $ cd features/
vscode ➜ ~/features (main) $ git checkout terraform-gpg
Branch 'terraform-gpg' set up to track remote branch 'terraform-gpg' from 'origin'.
Switched to a new branch 'terraform-gpg'
vscode ➜ ~/features (terraform-gpg) $ sudo su -
root ➜ ~ $ cd /home/vscode/features/src/terraform/
root ➜ .../vscode/features/src/terraform $ ./install.sh 
TERRAFORM_VERSION=1.3.9
TFLINT_VERSION=0.45.0
TERRAGRUNT_VERSION=0.44.0
Downloading terraform...
(*) Downloading GPG key...
gpg: keyserver receive failed: Connection timed out
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: Connection timed out
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: Connection timed out
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: Connection timed out
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: Connection timed out
(*) Failed getting key, retring in 10s...
(!) Failed to get gpg key.
root ➜ .../vscode/features/src/terraform $ 
```

Reply:
Manually by specifying HTTPPROXY.

I would like you to do something with the HTTP_PROXY setting, not the HTTPPROXY variable.

```
root ➜ .../vscode/features/src/terraform $ HTTPPROXY=$HTTP_PROXY ./install.sh
TERRAFORM_VERSION=1.3.9
TFLINT_VERSION=0.45.0
TERRAGRUNT_VERSION=0.44.0
Downloading terraform...
(*) Downloading GPG key...
gpg: Signature made Wed Feb 15 17:25:31 2023 UTC
gpg:                using RSA key 374EC75B485913604A831CC7C820C6D5CD27AB87
gpg: Good signature from "HashiCorp Security (hashicorp.com/security) <security@hashicorp.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: C874 011F 0AB4 0511 0D02  1055 3436 5D94 72D7 468F
     Subkey fingerprint: 374E C75B 4859 1360 4A83  1CC7 C820 C6D5 CD27 AB87
terraform_1.3.9_linux_amd64.zip: OK
Archive:  terraform_1.3.9_linux_amd64.zip
  inflating: terraform               
Downloading tflint...
gpg: Signature made Sun Feb 12 16:04:43 2023 UTC
gpg:                using RSA key 1780244FBAEB62C74476BE498CE69160EB3F2FE9
gpg: Good signature from "Kazuma Watanabe <watassbass@gmail.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 2DA7 A4B1 1347 B217 3852  31D1 131A 2054 C7B3 FB65
     Subkey fingerprint: 1780 244F BAEB 62C7 4476  BE49 8CE6 9160 EB3F 2FE9
tflint_linux_amd64.zip: OK
Archive:  /tmp/tf-downloads/tflint_linux_amd64.zip
  inflating: tflint                  
Downloading Terragrunt...
terragrunt_linux_amd64: OK
Done!
root ➜ .../vscode/features/src/terraform $ terraform --version
Terraform v1.3.9
on linux_amd64
root ➜ .../vscode/features/src/terraform $ 
```